Packet++/src/SomeIpSdLayer.cpp

```diff
@@ -711,7 +711,7 @@ bool SomeIpSdLayer::addOptionIndex(uint32_t indexEntry, uint32_t indexOffset)
 		return true;
 	}
 
-	if (indexFirstOption + lenFirstOption + 1 == indexOffset)
+	if (static_cast<uint8_t>(indexFirstOption + lenFirstOption + 1) == indexOffset)
```

Comment:
`indexOffset` is `uint32_t`, so maybe it's better to cast to `uint32_t`?

It's also safer because we add `uint8_t` numbers which may overflow the size of `uint8_t` but will never overflow `uint32_t`